qt/scientific_interfaces/Muon/MuonAnalysisFitDataPresenter.cpp

```diff
@@ -303,7 +303,15 @@ std::vector<std::string> MuonAnalysisFitDataPresenter::generateWorkspaceNames(
   const auto periods = m_dataSelector->getPeriodSelections();
 
   Muon::DatasetParams params;
-  const std::string instRuns = instrument + runString;
+  std::string runNumber = runString;
+  auto index = runString.find(instrument);
```

Comment:
Why not add this code to `parseRunLabel` instead?

Reply:
I just had a go at moving and sometimes the instrument is blank in parse run label and it therefor falls over if a path is given